Fakebook.Profile/Fakebook.Profile.DataAccess/EntityModel/EntityProfile.cs

```diff
@@ -1,50 +1,123 @@
+        {
+            Email = email;
+            FirstName = firstName;
+            LastName = lastName;
+            BirthDate = dob;
+            PhoneNumber = phoneNumber;
+            FollowerEmails = followerEmails;
+            FollowingEmails = followingEmails;
+            ProfilePictureUrl = new Uri(ProfileConfiguration.DefaultUri);
+        }
+
         /// <summary>
-        /// The primary key, the user's email.
+        /// Create a profile with user-specified profile picture
+        /// </summary>
+        /// <param name="email">The user's email. Must be unique.</param>
+        /// <param name="firstName">The user's firstname.</param>
+        /// <param name="lastName">The user's lastname.</param>
+        /// <param name="dob">The user's date of birth.</param>
+        /// <param name="phoneNumber">The user's phone number.</param>
+        /// <param name="status">The user's status.</param>
+        /// <param name="profilePictureUrl">The url of the user's profile picture.</param>
+        public EntityProfile(string email, string firstName, string lastName, DateTime dob, Uri profilePictureUrl, string? phoneNumber = null, string? status = null, IList<string>? followerEmails = null, IList<string>? followingEmails = null)
+        {
+            Email = email;
+            FirstName = firstName;
+            LastName = lastName;
+            BirthDate = dob;
+            PhoneNumber = phoneNumber;
+            FollowerEmails = followerEmails;
+            FollowingEmails = followingEmails;
+            ProfilePictureUrl = profilePictureUrl;
+        }
+
+        /// <summary>
+        /// The user's ID.  The primary key.  Cannot be null.
         /// </summary>
         [Key]
-        [Column(name: nameof(Email))]
+        [Required]
+        public int UserId { get; set; }
+
+        /// <summary>
+        /// The user's unique email. Cannot be null.
+        /// </summary>
+        [Required]
+        [MaxLength(45),MinLength(5)]
         public string Email { get; set; }
 
+        /// <summary>
+        /// The user's followers. Can be null.
+        /// </summary>
+        [NotMapped]
+        public IList<string>? FollowerEmails { get; set; } 
+
+        /// <summary>
+        /// The user's followees. Can be null.
+        /// </summary>
+        [NotMapped]
+        public IList<string>? FollowingEmails { get; set; } 
+
         /// <summary>
         /// The Uri of the user's profile picture. Can be null.
         /// </summary>
-        public Uri ProfilePictureUrl { get; set; }
+        public Uri? ProfilePictureUrl { get; set; }
 
         /// <summary>
         /// The user's first name. Cannot be null.
         /// </summary>
+        [MaxLength(25)]
         public string FirstName { get; set; }
 
         /// <summary>
-        /// The user's last name, cannot be null.
+        /// The user's last name. Cannot be null.
         /// </summary>
+        [Required]
+        [MaxLength(25),MinLength(2)]
```

Comment:
likewise names, it doesn't really provide value to restrict them beyond [Required], just in case.